tests/s3/versioning/test_versioning_multipart_part_2.py

```diff
@@ -0,0 +1,228 @@
+        self.log.info("Step 3: Initiate multipart upload.")
+        res = self.s3mp_test_obj.create_multipart_upload(self.bucket_name, self.object_name)
+        assert_utils.assert_true(res[0], res[1])
+        mpu_id = res[1]["UploadId"]
+        self.log.info("Multipart Upload initiated with mpu_id %s", mpu_id)
+        self.log.info("Step 4: Upload parts.")
+        self.log.info("Uploading parts into bucket")
+        res = self.s3mp_test_obj.upload_parts(mpu_id, self.bucket_name, self.object_name,
+                                              multipart_obj_size=100, total_parts=10,
+                                              multipart_obj_path=self.test_file_path)
+        assert_utils.assert_true(res[0], res[1])
+        assert_utils.assert_equal(len(res[1]), 10, res[1])
+        self.log.info("Step 5: Abort multipart upload.")
+        res = self.s3mp_test_obj.abort_multipart_upload(self.bucket_name, self.object_name, mpu_id)
+        assert_utils.assert_true(res[0], res[1])
+        self.log.info("Step 6: Perform List Object Versions.")
+        s3ver_cmn_lib.check_list_object_versions(
+            self.s3ver_test_obj, bucket_name=self.bucket_name, expected_versions={})
+        self.log.info("Step 7: Perform GET/HEAD Object on aborted object.")
+        s3ver_cmn_lib.check_get_head_object_version(self.s3test_obj, self.s3ver_test_obj,
+                                                    self.bucket_name, self.object_name,
+                                                    get_error_msg=err_msg.NO_SUCH_KEY_ERR,
+                                                    head_error_msg=err_msg.NOT_FOUND_ERR)
+        self.log.info("ENDED: Test Abort Multipart Upload does not create a new version.")
+
+    @pytest.mark.parallel
+    @pytest.mark.s3_versioning
+    @pytest.mark.s3_ops
+    @pytest.mark.tags('TEST-41289')
+    def test_upload_multiple_versions_to_multipart_uploaded_object_in_versioned_bucket_41289(self):
+        """Test Upload multiple versions to a multipart uploaded object in a versioned bucket."""
+        self.log.info("STARTED: Test Upload multiple versions to a multipart uploaded object in "
+                      "a versioned bucket.")
+        versions = {}
+        self.log.info("Step 1: Create bucket.")
+        assert_utils.assert_in(self.bucket_name, self.bkt_list,
+                               f"Bucket '{self.bucket_name}' does not exists")
+        self.log.info("Step 2: PUT Bucket versioning with status as Enabled.")
+        self.s3ver_test_obj.put_bucket_versioning(self.bucket_name, status="Enabled")
+        self.log.info("Step 3: Upload a multipart object - mpobject1: mpversionid1")
+        s3ver_cmn_lib.upload_version(self.s3mp_test_obj, self.bucket_name, self.object_name,
+                                     self.test_file_path, versions_dict=versions,
+                                     is_multipart=True, total_parts=10, file_size=100)
```

Comment:
We really dont need to have bigger object sizes, I think 10mb is also okay for these scenarios

Reply:
100MB is not big size for Multipart scenario so I think this is okay.